=== modified file 'Bugzilla/DB/Schema.pm'
--- Bugzilla/DB/Schema.pm	2012-02-15 15:26:34 +0000
+++ Bugzilla/DB/Schema.pm	2012-02-16 15:40:05 +0000
@@ -231,6 +231,7 @@
         isactive => {TYPE => 'BOOLEAN', NOTNULL => 1,
                      DEFAULT => 'TRUE'},
         visibility_value_id => {TYPE => 'INT2'},
+        optgroup_id => {TYPE => 'INT2'},
     ],
     # Note that bz_add_field_table should prepend the table name
     # to these index names.

=== modified file 'template/en/default/admin/custom_fields/edit.html.tmpl'
--- template/en/default/admin/custom_fields/edit.html.tmpl	2010-08-27 03:27:08 +0000
+++ template/en/default/admin/custom_fields/edit.html.tmpl	2012-01-30 11:44:33 +0000
@@ -140,7 +140,9 @@
         <th>&nbsp;</th>
         <td>
           <a href="editvalues.cgi?field=[% field.name FILTER uri %]">Edit
-            legal values for this field</a>.
+            legal values for this field</a>.<br>
+          <a href="page.cgi?id=fieldgroups.html&field=[% field.name FILTER uri %]">Edit
+            legal groups for this field</a>.
         </td>
 
         <th>

=== modified file 'template/en/default/admin/fieldvalues/create.html.tmpl'
--- template/en/default/admin/fieldvalues/create.html.tmpl	2010-02-02 01:34:26 +0000
+++ template/en/default/admin/fieldvalues/create.html.tmpl	2012-01-30 11:44:33 +0000
@@ -86,6 +86,7 @@
         </td>
       </tr>
     [% END %]
+    [% Hook.process('end_parameters') %]
   </table>
   <input type="submit" id="create" value="Add">
   <input type="hidden" name="action" value="new">

=== modified file 'template/en/default/admin/fieldvalues/edit.html.tmpl'
--- template/en/default/admin/fieldvalues/edit.html.tmpl	2010-05-14 14:25:05 +0000
+++ template/en/default/admin/fieldvalues/edit.html.tmpl	2012-01-30 11:44:33 +0000
@@ -100,6 +100,7 @@
            [% END %]
       </td>
     </tr>
+    [% Hook.process('end_parameters') %]
   </table>
   <input type="hidden" name="value" value="[% value.name FILTER html %]">
   <input type="hidden" name="action" value="update">

=== modified file 'template/en/default/admin/fieldvalues/list.html.tmpl'
--- template/en/default/admin/fieldvalues/list.html.tmpl	2010-07-22 22:46:02 +0000
+++ template/en/default/admin/fieldvalues/list.html.tmpl	2012-01-30 11:44:33 +0000
@@ -50,6 +50,10 @@
        name => "sortkey"
        heading => "Sortkey"
      },
+     { 
+       name => "group"
+       heading => "Group"
+     },
      {
        name => "isactive"
        heading => "Enabled for $terms.bugs"
@@ -83,7 +87,11 @@
        }
     %]
   [% END %]
-
+    [% overrides.group.name.${check_value.name} = {
+         override_content => 1
+         content => "$check_value.optgroup.name"
+       }
+    %]
 [% END %]
 
 [% PROCESS admin/table.html.tmpl

=== modified file 'template/en/default/bug/create/create.html.tmpl'
--- template/en/default/bug/create/create.html.tmpl	2012-02-28 22:28:17 +0000
+++ template/en/default/bug/create/create.html.tmpl	2012-04-02 07:45:27 +0000
@@ -493,7 +493,7 @@
     [% NEXT UNLESS field.enter_bug %]
     [% SET value = ${field.name}.defined ? ${field.name} : "" %]
     <tr [% 'class="expert_fields"' IF !field.is_mandatory %]>
-      [% INCLUDE bug/field.html.tmpl 
+      [% INCLUDE fieldgroups/field.html.tmpl 
         bug = default, field = field, value = value, editable = 1, 
         value_span = 3 %]
     </tr>

=== modified file 'template/en/default/bug/edit.html.tmpl'
--- template/en/default/bug/edit.html.tmpl	2011-12-05 18:58:12 +0000
+++ template/en/default/bug/edit.html.tmpl	2012-01-30 15:11:35 +0000
@@ -919,7 +919,7 @@
   [% USE Bugzilla %]
   [% FOREACH field = Bugzilla.active_custom_fields %]
     <tr>
-      [% PROCESS bug/field.html.tmpl value = bug.${field.name}
+      [% PROCESS fieldgroups/field.html.tmpl value = bug.${field.name}
                                      editable = bug.check_can_change_field(field.name, 0, 1)
                                      value_span = 2 %]
     </tr>

=== modified file 'template/en/default/global/messages.html.tmpl'
--- template/en/default/global/messages.html.tmpl	2011-11-26 00:13:18 +0000
+++ template/en/default/global/messages.html.tmpl	2012-01-30 11:44:33 +0000
@@ -328,6 +328,10 @@
           <li>Sortkey updated to 
             <em>[% changes.sortkey.1 FILTER html %]</em>.</li>
         [% END %]
+        [% IF changes.optgroup_id %]
+        	<li>Value now belongs to option group
+        	  <em>[% value.optgroup.name FILTER html %]</em>.</li>
+        [% END %]
         [% IF changes.visibility_value_id %]
           [% IF value.visibility_value.defined %]
             <li>It only appears when 

